test/fakeintake/client/client_integration_test.go

```diff
@@ -67,74 +42,66 @@ func TestIntegrationClient(t *testing.T) {
 		assert.Equal(t, http.StatusOK, resp.StatusCode)
 
 		client := NewClient(fi.URL())
-		// max wait for 250 ms
-		err = backoff.Retry(client.GetServerHealth, backoff.WithMaxRetries(backoff.NewConstantBackOff(10*time.Millisecond), 25))
-		require.NoError(t, err, "Failed waiting for fakeintake")
 
-		payloads, err := client.getFakePayloads(testEndpoint)
-		assert.NoError(t, err, "Error getting payloads")
-		assert.Equal(t, 1, len(payloads))
-		assert.Equal(t, "totoro|5|tag:valid,owner:pducolin", string(payloads[0].Data))
-		assert.Equal(t, "text/plain", payloads[0].Encoding)
-		assert.Equal(t, mockClock.Now().UTC(), payloads[0].Timestamp)
+		stats, err := client.RouteStats()
+		require.NoError(t, err, "Error getting payloads")
+		expectedStats := map[string]int{
+			"/foo/bar": 1,
+		}
+		assert.Equal(t, expectedStats, stats)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏

IIUC we now test that we have the expected number of payload on the endpoint. But we do not test for the content anymore. I think the test is less precise now. But maybe we do not really care and this is enough. WDYT?

Reply:
Indeed content is tested on the server test side, here we test the integration, but I should add an integration test with a test fixture payloads that the client can read